### PR TITLE
Timesheets > Select/Deselect All, additional feature about select or …

### PR DESCRIPTION
**Pull request by baonguyengia24**
An extra function for choosing or deselecting all. When you click choose all, it selects all of the checkboxes in the row, and vice versa. When you pick a checkbox, the imdeterminate property is applied.

### CODE REVIEW

**Comment by baonguyengia24** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153714985/c5c01328-427e-400f-85b3-404713b396dc

File: angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts [hunk 1]
```diff
@@ -418,6 +419,39 @@ export class TimesheetDetailComponent extends PagedListingComponentBase<Timeshee
     this.contextMenuPosition.y = e.clientY + 'px';
     this.menu.openMenu();
   }
+
+  SelectAllChanged(event){
+    if(event.checked){
+      this.TimesheetDetaiList = this.TimesheetDetaiList.map((item) => {
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts]: tìm hiểu kỹ xem map trong js chạy ntn rồi tối ưu lại đoạn code này.
đừng gán lại this.TimesheetDetaiList

File: angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts [hunk 2]
```diff
@@ -418,6 +419,39 @@ export class TimesheetDetailComponent extends PagedListingComponentBase<Timeshee
...
+    }
+  }
+
+  SelectChanged(event){
+    if(this.TimesheetDetaiList.every((item) => item.isChecked === true)){
+      this.indeterminate = false;
+      this.checkSelectAllBox = true;
+    }
+    else {
+      if(this.TimesheetDetaiList.filter((item) => item.isChecked).length >0 ){
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts]: dưới này cũng thế, tìm hiểu kỹ các funtion của array trong js mà dùng cho tối ưu

**Review by tiennguyenhuuncc — COMMENTED** [review]
check comment

**Comment by baonguyengia24** [comment]
New update after review
https://github.com/ncc-erp/ncc-erp-project/assets/153714985/650d73c5-6fb3-4c9c-b8ad-49c4fc812763

**Comment by baonguyengia24** [comment]
New update after review
https://github.com/ncc-erp/ncc-erp-project/assets/153714985/ee020b85-5656-4b54-aeab-9ffe8a7ee294

**Comment by baonguyengia24** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153714985/c4b3162e-6205-430a-8e2a-4311923cfde5

**Review by tiennguyenhuuncc — APPROVED** [review]

**Pull request merged**